apps/dashboard/src/components/activity/activity-job-item.tsx

```diff
@@ -0,0 +1,215 @@
+          (job.digest as IDigestRegularMetadata)?.unit ?? ''
+        }`;
+      }
+      if (job.status === 'delayed') {
+        return `Collecting Digest events for ${(job.digest as IDigestRegularMetadata)?.amount ?? 0} ${
+          (job.digest as IDigestRegularMetadata)?.unit ?? ''
+        }`;
+      }
+      return 'Digest failed';
+
+    case 'delay':
+      if (job.status === 'completed') {
+        return 'Delay completed';
+      }
+
+      if (job.status === 'delayed') {
+        return (
+          'Waiting for ' +
+          (job.digest as IDelayRegularMetadata)?.amount +
+          ' ' +
+          (job.digest as IDelayRegularMetadata)?.unit
+        );
+      }
+
+      return 'Delay failed';
+
+    default:
+      if (job.status === 'completed') {
+        return 'Message sent successfully';
+      }
+      if (job.status === 'pending') {
+        return 'Sending message';
+      }
+      return 'Message delivery failed';
+  }
+}
+
+function getJobIcon(type?: StepTypeEnum) {
+  const Icon = STEP_TYPE_TO_ICON[type?.toLowerCase() as keyof typeof STEP_TYPE_TO_ICON] ?? Route;
+
+  return <Icon className="h-3.5 w-3.5" />;
+}
+
+function getJobColor(status: string) {
+  switch (status) {
+    case 'completed':
+      return 'success';
+    case 'failed':
+      return 'destructive';
+    case 'delayed':
+      return 'warning';
+    case 'merged':
+      return 'neutral-300';
+    default:
+      return 'neutral-300';
+  }
+}
+
+function JobDetails({ job }: { job: IActivityJob }) {
+  return (
+    <div className="border-t border-neutral-100 p-4">
+      <div className="flex flex-col gap-4">
+        {job.executionDetails && job.executionDetails.length > 0 && (
+          <div className="flex flex-col gap-2">
+            {job.executionDetails.map((detail, index) => (
+              <ExecutionDetailItem key={index} detail={detail} />
+            ))}
+          </div>
+        )}
+        {/*
+        TODO: Missing backend support for digest events widget
+        {job.type === 'digest' && job.digest?.events && (
+          <ActivityDetailCard title="Digest Events" expandable={true} open>
+            <div className="min-w-0 max-w-full font-mono">
+              {job.digest.events.map((event: DigestEvent, index: number) => (
+                <div key={index} className="group flex items-center gap-2 rounded-sm px-1 py-1.5 hover:bg-neutral-100">
+                  <RiCheckboxCircleLine className="text-success h-4 w-4 shrink-0" />
+                  <div className="flex items-center gap-2 truncate">
+                    <span className="truncate text-xs text-neutral-500">{event.type}</span>
+                    <span className="text-xs text-neutral-400">
+                      {`${format(new Date(job.updatedAt), 'HH:mm')} UTC`}
+                    </span>
+                  </div>
+                </div>
+              ))}
+            </div>
+          </ActivityDetailCard>
+        )} */}
```

Comment:
Currently backend support is lacking the ability to show the transactionIds of the digested events and we only store the payload there